code/obj/machinery/lightswitch.dm

```diff
@@ -14,16 +14,17 @@
 	var/area/area = null
 	var/otherarea = null
 	//	luminosity = 1
-	// mats = 6 fuck you mport
+	mats = list("MET-1"=10,"CON-1"=15)
 	var/datum/light/light
 
 
 /obj/machinery/light_switch/New()
 	..()
 	UnsubscribeProcess()
-	light = new /datum/light/point
+	if (!light)
+		light = new /datum/light/point
 	SPAWN(0.5 SECONDS)
-		src.area = src.loc.loc
+		src.area = get_area(src)
```

Comment:
Not that you added that spawn, but I think this means that there's a half second window for runtimes to happen because area isn't set when toggling (that's normally covered up by the pregame) or toggling the previously assigned area.

I'm kinda feeling like calling New() in was_built_from_frame isn't great in general (kinda inelegant for deconned and re-built switches, a double call for completely new switches), but I'm not sure if that's actually an issue or if it's just me going "I personally wouldn't code it like that so it sucks".

Reply:
Yeeah `src.New()` gives me bad vibes